feat: update 100 (#148)
tari-project/tari-dot-com: _updates/2023-02-17-update-100.md

What's so special about StageNet, you ask? Well, after several iterations of testnets with cool names like our currently running Esme, Igor, and our previously retired Dibbler, Weatherwax, Ridcully, and Stibbons (RIP ✌️), we're approaching a time that requires a stable testnet that mirrors our future mainnet as closely as possible. And that's what StageNet is all about. It's a long-lived, stable testnet that won't reset often, because we know how annoying that can be. You can experiment with the network and test new features risk-free just as they should run on Mainnet. 

I know what you're thinking. Without Mainnet, what's the difference between this stable testnet and the previous ones? Good question! And here's the answer: It's all in how we're caring for it. We're transitioning to an 8-week release cycle schedule! Yes, you heard it right. We may not have a mainnet yet, but we're gonna start developing like we do. So, StageNet will get new updates every 8 weeks, just like the mainnet will someday soon™️. This is a big change for our protocol team's workflow, and we're excited to iron out the kinks early to ensure a smoother mainnet transition.

> 

Reviewer: No issues with the light, conversational style, but dial it back 2 notches.

### How this will look in the immediate future:

> Currently, the Protocol team is a highly functioning group, throwing out new features and fixes whenever inspiration strikes. When work gets completed it gets merged into `development`. Then, once every so often we decide "hey we've done some really great work here, we should share it with people". And somebody will tag a release. It's gotten us to where we are but can be unpredictable. So we're introducing a little more structure to our process.

Reviewer: ```suggestion
Currently, the Protocol team is a highly functioning group, throwing out new features and fixes whenever inspiration strikes. When work gets completed it gets merged into the `development` branch on our [Github repo](link). 

Then, every so often we decide "hey we've done some really great work here, we should share it with people". And somebody will tag a release. This approach has gotten us to where we are but is a little unpredictable. So we're introducing a little more structure to our process.
```

Author: Updated with suggestion.

> Going forward, we'll be taking a more meticulous approach to code selection and testing. We'll be running features through the gauntlet of our named testnets to ensure they're truly ready for prime time. No half-baked ideas making their way through, only once a feature has passed the rigorous testing, will it earn its place in the coveted stagenet branch. These features will be pruned and picked from our `developmnet` branch into the `stagenet` branch. Only then will a tag be created from this branch, and released for your ultimate enjoyment.

Reviewer: Consistent naming

```suggestion
Going forward, we'll be taking a more meticulous approach to code selection and testing. We'll be running features through the gauntlet of our named testnets to ensure they're truly ready for prime time. No half-baked ideas making their way through, only once a feature has passed the rigorous testing, will it earn its place in the coveted StageNet branch. These features will be pruned and picked from our `developmnet` branch into the `stagenet` branch. Only then will a tag be created from this branch, and released for your ultimate enjoyment.
```

Author: Updated the naming inconsistencies. Also did a scan for some other inconsistent usages "testnets" vs "test networks" and "Mainnet" vs "mainnet"

---
layout: update
tag: Developer Update
date: 2023-02-17
author: brianp
thumbnail: update-background.jpg
title: "Tari Base Node v0.45.0 Released on StageNet"
"
class: subpage
---

> ## Tari Protocol 0.45.0 Released: Your Wait for Mainnet Continues

Reviewer: 🤔 
```suggestion
## Tari Protocol 0.45.0 Released: Approaching mainnet one minor release at a time
```

Author: 😈

It's that time again! The Tari protocol team is excited to announce the release of version 0.45.0, bringing a number of new features and improvements to the network. This release also marks an important milestone in our development with big news - wait for it - we've launched the Tari StageNet! Yes, you heard it right. We know, you're excited about the Mainnet, but trust me, this is almost as good.

### What's so special about StageNet?
After several iterations of testnets with cool names like our currently running Esme, Igor, and our previously retired Dibbler, Weatherwax, Ridcully, and Stibbons (RIP ✌️), we're approaching a time that requires a stable testnet that mirrors our future Mainnet as closely as possible. And that's what StageNet is all about. It's a long-lived, stable testnet that won't reset often, because we know how annoying that can be. You can experiment with the network and test new features risk-free just as they should run on Mainnet.

> We won't be getting rid of our named testnets though! They still play an integral part in the development of the Tari network. This is where features are tested a little more fast and lose. We'll do our best not to, but these networks may see breakage sometimes. Or a new one may crop up to test a specific subset of features. For example right now our two running named testnets exist for different reasons. Our `Esme` network has been our stable-er testnet prior to `StageNet`. It sees new features and enhancements as they come out, and supports our mobile wallets [Tari Aurora](https://aurora.tari.com/) on [iOS](https://apps.apple.com/us/app/tari-aurora/id1503654828) and [Android](https://play.google.com/store/apps/details?id=com.tari.android.wallet). Where our `Igor` testnet is used in the development of our [DAN Layer](https://www.tari.com/updates/2023-02-01-update-99.html) and sees frequent resets to help support the rapid pace of the development team.

Reviewer: ```suggestion
We won't be getting rid of our named testnets though. They still play an integral part in the development of the Tari network, and are where we can play a little more fast and loose with features. We'll do our best not to, but these networks may see breakage sometimes. Or a new one may crop up to test a specific subset of features. For example, right now our two running named testnets exist for different reasons. 

Our `Esme` network has been our stable-er testnet prior to `StageNet`. It sees new features and enhancements as they come out, and supports our mobile wallets [Tari Aurora](https://aurora.tari.com/) on [iOS](https://apps.apple.com/us/app/tari-aurora/id1503654828) and [Android](https://play.google.com/store/apps/details?id=com.tari.android.wallet). 

Our `Igor` testnet is used in the development of the [DAN Layer](https://www.tari.com/updates/2023-02-01-update-99.html) and sees frequent resets to help support the rapid pace of the development team.
```

> I know what you're thinking. Without Mainnet, what's the difference between this new stable StageNet and the previous ones? And here's the answer: It's all in how we're shepherding it. We're transitioning to an 8-week release cycle schedule! Yes, you heard it right. We may not have a Mainnet yet, but we're gonna start developing like we do. StageNet will get new updates every 8 weeks, just like the Mainnet will someday soon™️. This is a big change for our protocol team's workflow, and we're excited to iron out the kinks early to ensure a smoother Mainnet transition.

Reviewer: ```suggestion
I know what you're thinking. Without Mainnet, what's the difference between this new stable StageNet and the previous ones? And here's the answer: It's all in how we're shepherding it. We're transitioning to an 8-week release cycle schedule. Yes, you heard it right. We may not have a Mainnet yet, but we're gonna start developing like we do. StageNet will get new updates every 8 weeks, just like the Mainnet will soon™️. This is a big change for our protocol team's workflow, and we're excited to iron out the kinks early to ensure a smoother Mainnet transition.
```

### How this will look in the immediate future:

Currently, the Protocol team is a highly functioning group, throwing out new features and fixes whenever inspiration strikes. When work gets completed it gets merged into the `development` branch on our [Github repo](https://github.com/tari-project/tari/blob/development/changelog.md).

Then, every so often we decide "hey we've done some really great work here, we should share it with people". And somebody will tag a release. This approach has gotten us to where we are but is a little unpredictable. So we're introducing a little more structure to our process.

> Going forward, we'll be taking a more meticulous approach to code selection and testing. We'll be running features through the gauntlet of our named testnets to ensure they're truly ready for prime time. No half-baked ideas making their way through, only once a feature has passed the rigorous testing, will it earn its place in the coveted StageNet branch. These features will be pruned and picked from our `developmnet` branch into the `stagenet` branch. Only then will a tag be created from this branch, and released for your ultimate enjoyment.

Reviewer: ```suggestion
Going forward, we'll be taking a more meticulous approach to code selection and testing. We'll be running features through the gauntlet of our named testnets to ensure they're truly ready for prime time. No half-baked ideas muddling their way in, only once a feature has passed the rigorous testing will it earn its place in the coveted StageNet branch. These features will be pruned and picked from our `development` branch into the `stagenet` branch. Only then will a tag be created from this branch, and released for your ultimate enjoyment.
```
